fdbserver/RestoreApplier.actor.cpp

```diff
@@ -128,29 +128,29 @@ ACTOR static Future<Void> handleSendMutationVectorRequest(RestoreSendVersionedMu
 	state bool isDuplicated = true;
 	if (curMsgIndex.get() == req.msgIndex - 1) {
 		isDuplicated = false;
-		ASSERT(req.mutations.size() == req.mVersions.size());
 
-		for (int mIndex = 0; mIndex < req.mutations.size(); mIndex++) {
-			const MutationRef& mutation = req.mutations[mIndex];
-			const LogMessageVersion mutationVersion(req.mVersions[mIndex]);
+		for (int mIndex = 0; mIndex < req.versionedMutations.size(); mIndex++) {
+			const VersionedMutation& versionedMutation = req.versionedMutations[mIndex];
 			TraceEvent(SevFRMutationInfo, "FastRestoreApplierPhaseReceiveMutations", self->id())
 			    .detail("RestoreAsset", req.asset.toString())
-			    .detail("Version", mutationVersion.toString())
+			    .detail("Version", versionedMutation.version.toString())
 			    .detail("Index", mIndex)
-			    .detail("MutationReceived", mutation.toString());
-			batchData->counters.receivedBytes += mutation.totalSize();
-			batchData->counters.receivedWeightedBytes += mutation.weightedTotalSize(); // atomicOp will be amplified
+			    .detail("MutationReceived", versionedMutation.mutation.toString());
+			batchData->counters.receivedBytes += versionedMutation.mutation.totalSize();
+			batchData->counters.receivedWeightedBytes +=
+			    versionedMutation.mutation.weightedTotalSize(); // atomicOp will be amplified
 			batchData->counters.receivedMutations += 1;
-			batchData->counters.receivedAtomicOps += isAtomicOp((MutationRef::Type)mutation.type) ? 1 : 0;
+			batchData->counters.receivedAtomicOps +=
+			    isAtomicOp((MutationRef::Type)versionedMutation.mutation.type) ? 1 : 0;
 			// Sanity check
-			ASSERT_WE_THINK(req.asset.isInVersionRange(mutationVersion.version));
-			ASSERT_WE_THINK(req.asset.isInKeyRange(mutation));
+			ASSERT_WE_THINK(req.asset.isInVersionRange(versionedMutation.version.version));
+			ASSERT_WE_THINK(req.asset.isInKeyRange(versionedMutation.mutation));
 
 			// Note: Log and range mutations may be delivered out of order. Can we handle it?
-			batchData->addMutation(mutation, mutationVersion);
+			batchData->addMutation(versionedMutation.mutation, versionedMutation.version);
```

Comment:
Might be better to refactor `addMutation` to take `versionedMutation` as the parameter.